### PR TITLE
[FrameworkBundle] feature: add the ability to search for a route

### PR DESCRIPTION
**Pull request by Simperfit**
The PR add the documentation for core's PR symfony/symfony#26121

### CODE REVIEW

File: routing/debug.rst [hunk 1]
```diff
@@ -31,6 +31,20 @@ the route name as the command argument:
...
+.. code-block:: terminal
+
+    $ php bin/console debug:router blo
+    
+      Select one of the following routes to display its information:
+      [0] blog
+      [1] blog_show
+     >
+     
+If you choose ``blog_show`` it will show the route information.
```
Comment (wouterj) [inline routing/debug.rst]: What about merging this all in the previous code example, to reduce the amount of text/lines?

```rst
You can also get very specific information on a single route by including
the route name as the command argument:

.. code-block:: terminal

    $ php bin/console debug:router article_show
    # ...

    # or use part of the name to search for routes
    $ php bin/console debug:router blo
    
      Select one of the following routes to display its information:
      [0] blog
      [1] blog_show
     >
```

**Comment by wouterj** [comment]
I like the feature! :+1: for when it gets merged

File: routing/debug.rst (hunk 1 above)
Comment (Simperfit) [reply routing/debug.rst]: updated

**Comment by Simperfit** [comment]
PR is ready, updated with the final merged PR on the core.

File: routing/debug.rst (hunk 1 above)
Comment (Simperfit) [reply routing/debug.rst]: what sentence do you want to drop, the one explaining or the one in the comments?

Comment (xabbuh) [reply routing/debug.rst]: this one: "If you choose ``blog_show`` it will show the route information."

Comment (Simperfit) [reply routing/debug.rst]: done

**Review by xabbuh — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Hamza.

**Pull request merged**